x509util/certificate.go

```diff
@@ -83,6 +83,83 @@ func NewCertificate(cr *x509.CertificateRequest, opts ...Option) (*Certificate,
 	return &cert, nil
 }
 
+// FromX509Certificate converts an x509.Certificate to a Certificate such that
+// calling Certificate.GetCertificate() will return a copy of the original
+// x509.Certificate.
+func FromX509Certificate(cert *x509.Certificate) *Certificate {
+	c := new(Certificate)
+
+	// Unparsed data
+	c.PublicKey = cert.PublicKey
+	c.PublicKeyAlgorithm = cert.PublicKeyAlgorithm
+
+	// Subject
+	c.Subject = newSubject(cert.Subject)
+
+	// SAN values
+	c.DNSNames = cert.DNSNames
+	c.EmailAddresses = cert.EmailAddresses
+	c.IPAddresses = cert.IPAddresses
+	c.URIs = cert.URIs
```

Comment:
We should add the SANs too, as the device attestation certificates use another type of SANs by default. To do this, and probably to set the right DNSNames, EmailsAddresses, IPAddresses and URIs we will need to decode the subject alternative name extension (`2.5.29.17`).